sympy/physics/paulialgebra.py

```diff
@@ -204,28 +204,42 @@ def evaluate_pauli_product(arg):
         start = end
 
         tmp = start.as_coeff_mul()
-        sigma_product = 1
+        sigma_products_by_label = {}
         com_product = 1
         keeper = 1
 
+        def add_sigma_product(p):
+            if (sigma_products_by_label.get(p.label)):
+                sigma_products_by_label[p.label] *= p
+            else:
+                sigma_products_by_label[p.label] = p
+
+        def flatten_sigma_product():
+            sigma_product = 1
+            nonlocal sigma_products_by_label
+            for label in sorted(sigma_products_by_label):
+                sigma_product *= sigma_products_by_label[label]
+
+            sigma_products_by_label = {}
```

Comment:
What is the purpose of the `nonlocal`?

It seems like it should just be a local variable.